test/functional/session_leak_test.js

```diff
@@ -29,36 +28,36 @@ beforeEach('Session Leak Before Each - setup session tracking', function() {
   const _acquire = ServerSessionPool.prototype.acquire;
   sandbox.stub(ServerSessionPool.prototype, 'acquire').callsFake(function() {
     const session = _acquire.apply(this, arguments);
-    activeSessions.add(session.id);
-    // console.log(`Active + ${JSON.stringify(session.id)} = ${activeSessions.size}`);
+    activeSessions.add(sessionId(session.id));
     return session;
   });
 
   const _release = ServerSessionPool.prototype.release;
   sandbox.stub(ServerSessionPool.prototype, 'release').callsFake(function(session) {
-    const id = session.id;
-    activeSessions.delete(id);
-    // console.log(`Active - ${JSON.stringify(id)} = ${activeSessions.size}`);
-    pooledSessions.add(id);
-    // console.log(`Pooled + ${JSON.stringify(id)} = ${activeSessions.size}`);
+    activeSessions.delete(sessionId(session.id));
+    pooledSessions.add(sessionId(session.id));
+
     return _release.apply(this, arguments);
   });
 
+  const _endAllPooledSessions = ServerSessionPool.prototype.endAllPooledSessions;
+  sandbox.stub(ServerSessionPool.prototype, 'endAllPooledSessions').callsFake(function() {
```

Comment:
Is this a bug in our leak detection, or a bug in our actual session management?

Reply:
It could be either, but when you added `endAllPooledSession` it has an extra hard reset of the pooled sessions. I suppose `endAllPooledSessions` could `release` all the sessions and then reset and it would make the existing stubs work as expected.